hscommon/hsproject/project.go

```diff
@@ -250,90 +254,91 @@ func (p *Project) searchPathEntries(pathEntry *hscommon.PathEntry, path string)
 	return nil
 }
 
-// CreateNewFolder creates a new directory
-func (p *Project) CreateNewFolder(path *hscommon.PathEntry) {
-	basePath := path.FullPath
-
-	filePathFormat := filepath.Join(basePath, "untitled%d")
-
-	var fileName string
-
-	for i := 0; ; i++ {
-		possibleFileName := fmt.Sprintf(filePathFormat, i)
-		if _, err := os.Stat(possibleFileName); os.IsNotExist(err) {
+func checkIfPathExists(fmtPath string, maxAttempt int) (fileName string, err error) {
+	for i := 0; i <= maxAttempt; i++ {
+		possibleFileName := fmt.Sprintf(fmtPath, i)
+		if _, err = os.Stat(possibleFileName); os.IsNotExist(err) {
 			fileName = possibleFileName
 
 			break
 		}
+	}
 
-		if i > maxProjectsCount {
-			dialog.Message("Could not create a new project folder!").Error()
+	if fileName == "" {
+		err = errors.New("could not create a new project file")
+	}
 
-			return
-		}
+	return fileName, err
+}
+
+// CreateNewFolder creates a new directory
+func (p *Project) CreateNewFolder(path *hscommon.PathEntry) (err error) {
+	basePath := path.FullPath
+
+	fmtPath := filepath.Join(basePath, "untitled%d")
+
+	fileName, err := checkIfPathExists(fmtPath, maxNewFileAttempts)
+	if err != nil {
+		dialog.Message(err.Error()).Error()
+
+		return err
 	}
 
-	if err := os.Mkdir(fileName, 0o644); err != nil {
-		dialog.Message("Could not create a new project folder!").Error()
+	err = os.Mkdir(fileName, newFileMode)
+	if err != nil {
+		dialog.Message(err.Error()).Error()
```

Comment:
Also need to log.

Reply:
5e2c480

We can clean this up further when we close #309